src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts

```diff
@@ -228,7 +228,12 @@ export class ServerTreeView extends Disposable implements IServerTreeView {
 			if (this._tree instanceof AsyncServerTree) {
 				const connectionParentGroup = this._tree.getElementById(newConnection.groupId) as ConnectionProfileGroup;
 				if (connectionParentGroup) {
-					connectionParentGroup.connections.push(newConnection);
+					const matchingConnectionIndex = connectionParentGroup.connections.findIndex((connection) => connection.matches(newConnection));
+					if (matchingConnectionIndex !== -1) {
+						connectionParentGroup.connections[matchingConnectionIndex] = newConnection;
+					} else {
+						connectionParentGroup.connections.push(newConnection);
```

Comment:
You should probably be using [ConnectionProfileGroup.addConnections](https://github.com/microsoft/azuredatastudio/blob/main/src/sql/platform/connection/common/connectionProfileGroup.ts#L145)

Make sure you go through the other places that you're directly adding the connections like here and verifying whether they should also be using addConnections as well.

Reply:
I will have to modify to replace in case of matching connections. The current logic doesn't work in case of this bug as the new connection we get has a completely different id from the existing connections.

Reply:
Yeah, it's randomly generated each time you create a new connection from the connection maker

Reply:
And even updating the connection if we found a match here isn't great, we should be disposing the original connection. That also seems like something that the connection group should be providing functions to do.